### PR TITLE
Move to v2-beta

### PR DESCRIPTION
**Pull request by joshwget**

### CODE REVIEW

File: executor/main.go [hunk 1]
```diff
@@ -18,20 +18,23 @@ func Main() {
...
 		"ping": func(event *events.Event, apiClient *client.RancherClient) error {
 			return nil
 		},
 	}
 
+	logrus.Info(os.Getenv("CATTLE_URL"))
+
 	router, err := events.NewEventRouter("rancher-compose-executor", 2000,
-		os.Getenv("CATTLE_URL"),
+		//os.Getenv("CATTLE_URL"),
```
Comment (joshwget) [inline executor/main.go]: This is always `http://localhost:8080/v1`. Not sure where this is being set.

Comment (ibuildthecloud) [inline executor/main.go]: In scripts somewhere.... But it's slightly valid use case.  You should rewrite the url as `/v2-beta`

File: executor/main.go [hunk 2]
```diff
@@ -18,20 +20,30 @@ func Main() {
...
 			return nil
 		},
 	}
 
+	url, err := url.Parse(os.Getenv("CATTLE_URL"))
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	if path.Base(url.Path) == "v1" {
```
Comment (joshwget) [inline executor/main.go]: Strip off `/v1` from the end of the URL here.

File: rancher/context.go [hunk 3]
```diff
@@ -166,14 +167,24 @@ func (c *Context) sanitizedProjectName() string {
...
 		}
 
-		if client, err := rancherClient.NewRancherClient(&rancherClient.ClientOpts{
-			Url:       c.Url,
+		url, err := url.Parse(c.Url)
+		if err != nil {
+			return nil, err
+		}
+
+		base := path.Base(url.Path)
```
Comment (joshwget) [inline rancher/context.go]: If the URL does not already end in `/v2-beta` or `/schemas`, append `/v2-beta` to the end.

**Pull request merged**